Change dependency between BOOT_BUILD_DIRECT_XIP_VARIANT and MCUboot bootloader mode

mkapala-nordic opened the pull request.

de-nordic commented on applications/nrf_desktop/README.rst:
```diff
@@ -1943,11 +1943,12 @@ Direct-xip mode
 Enable the ``CONFIG_BOOT_DIRECT_XIP`` Kconfig option in the bootloader configuration to make the MCUboot run the image directly from both image slots.
 The nRF Desktop's bootloader configurations do not enable the revert mechanism (``CONFIG_BOOT_DIRECT_XIP_REVERT``).
-When the direct-xip mode is enabled (the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option is set in the application configuration), the application modules that control the DFU transport do not request firmware upgrades and do not confirm the running image.
+When the direct-xip mode is enabled (the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` Kconfig option is set in the application configuration), the application modules that control the DFU transport do not request firmware upgrades and do not confirm the running image.
```
direct-xip ->Direct-XIP

mkapala-nordic replied:
Thanks. I think that we should align whole nrf_desktop docs to this convention, but I think we can do it possibly after rc1.

kapi-no replied:
It is called "direct-xip" in the MCUboot docs and that's why we used the same term here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/mcuboot/design.html#equal-slots-direct-xip

annwoj commented on applications/nrf_desktop/README.rst:
```diff
@@ -1943,11 +1943,12 @@ Direct-xip mode
 The direct-xip mode is used for the :ref:`background DFU <nrf_desktop_bootloader_background_dfu>`.
 In this mode, the MCUboot bootloader boots an image directly from a given slot, so the swap operation is not needed.
-Make sure to enable the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option in the application configuration to build application update images for both slots.
+Make sure to enable the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` Kconfig option in the application configuration to set the MCUboot mode of operations to Direct-XIP mode.
```
```suggestion
To set the MCUboot mode of operations to the Direct-XIP mode, make sure to enable the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` Kconfig option in the application configuration.
```

annwoj commented on applications/nrf_desktop/README.rst:
```diff
@@ -1943,11 +1943,12 @@ Direct-xip mode
 In this mode, the MCUboot bootloader boots an image directly from a given slot, so the swap operation is not needed.
-Make sure to enable the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` Kconfig option in the application configuration to build application update images for both slots.
+Make sure to enable the :kconfig:option:`CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` Kconfig option in the application configuration to set the MCUboot mode of operations to Direct-XIP mode.
+This option automatically enables the :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` to build application update images for both slots.
```
```suggestion
This option automatically enables :kconfig:option:`CONFIG_BOOT_BUILD_DIRECT_XIP_VARIANT` to build application update images for both slots.
```

annwoj reviewed (commented):
small suggestions added

kapi-no commented on samples/common/mcumgr_bt_ota_dfu/Kconfig:
```diff
@@ -138,6 +138,7 @@ config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_VALIDATION
 
 config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT
 	bool
+	depends on NCS_SAMPLE_MCUMGR_BT_OTA_DFU
```
Can you simply move this configuration option to the beginning of the `if NCS_SAMPLE_MCUMGR_BT_OTA_DFU` scope?

kapi-no commented on samples/common/mcumgr_bt_ota_dfu/Kconfig:
```diff
@@ -138,6 +138,7 @@ config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_VALIDATION
 
 config NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT
```
Can we remove the `imply MCUMGR_GRP_IMG_DIRECT_UPLOAD` imply according to the discussion with @de-nordic?

kapi-no reviewed (commented):
Looking good, left two comments

kapi-no commented:
I think you can squash last 3 commits into the "modules: invert" commit as this change and alignment of all affected projects should be atomic

mkapala-nordic commented:
Applied @kapi-no suggestions (squashed commits, changed from `Direct-XIP` to `direct-xip`, fixed `NCS_SAMPLE_MCUMGR_BT_OTA_DFU_MCUBOOT_DIRECT_XIP_SUPPORT`)

kapi-no commented:
We are planning to merge this PR. @de-nordic if you have any change requests we will address them in the follow-up PRs
